homeassistant/util/template.py

```diff
@@ -20,7 +25,11 @@ def render_with_possible_json_value(hass, template, value):
     except ValueError:
         pass
 
-    return render(hass, template, variables)
+    try:
+        return render(hass, template, variables)
+    except TemplateError:
+        _LOGGER.exception('Error parsing value')
+        return value
```

Comment:
Just curious, why not allow the exception to propagate and let the caller decide what to do with the exception?